rubin_nights/observatory_status.py

```diff
@@ -70,26 +96,34 @@ def get_dome_open_close(t_start: Time, t_end: Time, efd_client: InfluxQueryClien
                 + 1
             )
             open_start = opening.iloc[gaps].index.values
-            closing = dd.query("positionCommanded0 == 0 or positionCommanded1 == 0")
-            if len(closing) > 0:
-                # This means for an in-progress night with the dome open,
-                # or if the query doesn't include the closing time,
-                # we won't record the last open/close.
-                gaps = (
-                    np.concatenate(
-                        [
-                            np.array([-1]),
-                            np.where((np.diff(closing.index) / pd.Timedelta(1, "s")) > 5 * 60)[0],
-                        ]
-                    )
-                    + 1
+        closing = dome_shutter_close.query("day_obs == @day_obs")
+        if len(closing) > 0:
+            # This means for an in-progress night with the dome open,
+            # or if the query doesn't include the closing time,
+            # we won't record the last open/close.
+            gaps = (
+                np.concatenate(
+                    [
+                        np.array([-1]),
+                        np.where((np.diff(closing.index) / pd.Timedelta(1, "s")) > 5 * 60)[0],
```

Comment:
I'm confused by the `[0]` indexer at the end. That means that `gaps` is always a two element array, the first being 0, the second being the first close. What happens if the dome is opened and closed multiple times?

Reply:
This is not making it a two-element array, it's finding the actual indexes in the np.where. 
I can split this into two lines so it's clearer.